<launch>
    <group ns = "yolov7_ros">
        <node pkg="yolov7_ros" type="detect_ros.py" name="detect" output="screen" ns="yolov7">
        <!-- Download the official weights from the original repo -->
        <param name="weights_path" type="str"
        value="/home/sobits/catkin_ws/src/yolov7_ros/weights/best.pt"/>
        <!-- Path to a class_labels.txt file containing your desired class labels. The i-th entry corresponds to the i-th class id. For example, in coco class label 0 corresponds to 'person'. Files for the coco and berkeley deep drive datasets are provided in the 'class_labels/' directory. If you leave it empty then no class labels are visualized.-->
        <param name="classes_path" type="str" value="/home/sobits/catkin_ws/src/yolov7_ros/class_labels/yolov7_text.txt" />
        <!-- topic name to subscribe to -->
        <param name="img_topic" type="str" value="/rgb/image_raw" />
        <!-- topic name for the detection output -->
        <param name="out_topic" type="str" value="yolov7" />
        <!-- confidence threshold -->
        <param name="conf_thresh" type="double" value="0.85" />
        <!-- intersection over union threshold -->
        <param name="iou_thresh" type="double" value="0.45" />
        <!-- queue size for publishing -->
        <param name="queue_size" type="int" value="1" />
        <!-- image size to which to resize each input image before feeding into the
        network (the final output is rescaled to the original image size) -->
        <param name="img_size" type="int" value="640" />
        <!-- flag whether to also publish image with the visualized detections -->
        <param name="visualize" type="bool" value="true" />
        <!-- 'cuda' or 'cpu' -->
        <param name="device" type="str" value="cuda" />
    </node>    
    <!-- <group ns = "yolov7_ros"> -->
        <!--YOLOv3 3D-->
        <node name="object_pose_publisher_node" pkg="sobit_common_library" type="object_pose_publisher" output="screen"/>
        <param name="obj_under_rate" type="double" value="0.6"/>
        <param name="obj_top_rate" type="double" value="0.9"/>
        <param name="map_frame_name" type="string" value="/camera_link"/>
        <param name="base_frame_name" type="string" value="/camera_link"/>
        <param name="cloud_topic_name" type="string" value="/points2"/>
        <param name="max_distance_to_object" type="double" value="1.5"/>
        <param name="image_width" type="double" value="640"/>
    </group>
</launch>